crml/cennzx-spot/src/types.rs

```diff
@@ -52,15 +51,24 @@ impl Scaled for PerCent {
 	const SCALE: LowPrecisionUnsigned = 100;
 }
 
-decl_error! {
-	pub enum Error {
-		Overflow,
-		DivideByZero,
-		EmptyPool,
+#[derive(Debug)]
+pub enum Error {
+	Overflow,
+	DivideByZero,
+	EmptyPool,
+}
```

Comment:
Decided not to use `decl_error!` as it now becomes `Error<T>` and `T: Trait` bound needs to be added to every function that returns `Error::<T>::Overflow` etc. Instead, `Into` trait is implemented below to handle `&'static str` return type

Reply:
It would be good to keep `decl_error!` so the module can return error codes.
The internal conversion error types in this file should have some other internal error type which can convert into one of the `decl_error` variants.
But could leave a comment and do in another PR

Reply:
Yep I'll add a TODO